### PR TITLE
Make LiskAPI a class

### LINKED ISSUE #350: Make LiskAPI a class
Lovely ES6

### PR DESCRIPTION
**Pull request by Tschakki**
closes #350

### CODE REVIEW

File: src/api/liskApi.js [hunk 1]
```diff
@@ -51,175 +50,174 @@ const livePort = 8000;
...
+	 * @method broadcastSignedTransaction
+	 * @param transaction
+	 * @param callback
+	 *
+	 * @return API object
+	 */
+
+	broadcastSignedTransaction(
+		transaction, callback,
+	) {
```
Comment (willclarktech) [inline src/api/liskApi.js]: These will probably fit on one line now. (Also applies to other methods below.)

File: src/api/liskApi.js [hunk 2]
```diff
@@ -51,175 +50,174 @@ const livePort = 8000;
...
+		const request = {
+			requestUrl: `${privateApi.getFullURL.call(this)}/api/transactions`,
+			nethash: this.nethash,
+			requestParams: { transaction },
+		};
+
+		privateApi.sendRequestPromise.call(this, POST, request).then(result => callback(result.body));
```
Comment (willclarktech) [inline src/api/liskApi.js]: Perhaps beyond the scope of this PR but this method should match the `Promise`/callback pattern of `sendRequest`.

File: src/crypto/keys.js [hunk 3]
```diff
@@ -58,3 +58,20 @@ export function getPrivateAndPublicKeyFromSecret(secret) {
...
 }
+
+/**
+ * @method getAddressFromSecret
+ * @param secret
+ *
+ * @return keys object
+ */
+
+export function getAddressFromSecret(secret) {
```
Comment (willclarktech) [inline src/crypto/keys.js]: The name of this function is misleading. Maybe `getAddressAndPublicKeyFromSecret` or `getAccountFromSecret`? @Tosch110

File: src/crypto/keys.js [hunk 4]
```diff
@@ -58,3 +58,20 @@ export function getPrivateAndPublicKeyFromSecret(secret) {
 export function getKeys(secret) {
 	return getPrivateAndPublicKeyFromSecret(secret);
 }
+
+/**
+ * @method getAddressFromSecret
+ * @param secret
+ *
+ * @return keys object
```
Comment (willclarktech) [inline src/crypto/keys.js]: This is misleading.

File: test/api/liskApi.js [hunk 5]
```diff
@@ -110,7 +106,7 @@ describe('Lisk API module', () => {
 		});
 
 		it('should return a new instance of LiskAPI if called with a different context', () => {
-			LSK = LiskAPI.call({});
+			LSK = new LiskAPI({});
```
Comment (willclarktech) [inline test/api/liskApi.js]: This test no longer makes sense. It can be removed.

File: test/crypto/keys.js [hunk 6]
```diff
@@ -16,13 +16,18 @@ import {
...
 	getKeys,
+	getAddressFromSecret,
 } from '../../src/crypto/keys';
 import {
 	bufferToHex,
 } from '../../src/crypto/convert';
 
 describe('keys', () => {
 	const defaultSecret = 'secret';
+	const defaultAddress = {
```
Comment (willclarktech) [inline test/crypto/keys.js]: This should reuse the `expectedPublicKey` variable.

File: src/api/liskApi.js [hunk 7]
```diff
@@ -51,175 +50,174 @@ const livePort = 8000;
...
+
+		this.defaultTestnetPeers = options.peers || config.peers.testnet;
+
+		this.options = options;
+		this.ssl = options.ssl;
+		// Random peer can be set by settings with randomPeer: true | false
+		// Random peer is automatically enabled when no options.node has been entered. Else will be set
+		// to false.
+		// If the desired behaviour is to have an own node and automatic peer discovery, randomPeer
+		// should be set to true explicitly
```
[the remark this answers is not in the document]
Comment (Tosch110) [reply src/api/liskApi.js]: No, it is obsolete now

File: src/crypto/keys.js (hunk 3 above)
Comment (Tosch110) [reply src/crypto/keys.js]: Good idea, I would prefer getAddressAndPublicKeyFromSecret to be more accurate

File: test/crypto/keys.js [hunk 8]
```diff
@@ -16,20 +16,20 @@ import {
...
 } from '../../src/crypto/keys';
 import {
 	bufferToHex,
 } from '../../src/crypto/convert';
 
 describe('keys', () => {
 	const defaultSecret = 'secret';
+	const expectedPublicKey = '5d036a858ce89f844491762eb89e2bfbd50a4a0a0da658e4b2628b25b117ae09';
+	const expectedPrivateKey = '2bb80d537b1da3e38bd30361aa855686bde0eacd7162fef6a25fe97bf527a25b5d036a858ce89f844491762eb89e2bfbd50a4a0a0da658e4b2628b25b117ae09';
 	const defaultAddress = {
```
Comment (willclarktech) [inline test/crypto/keys.js]: Now that we're calling the function `getAddressAndPublicKeyFromSecret`, we should name this accordingly. Also let's make our use of `expectedVar` vs `defaultVar` consistent.

**Pull request merged**